src/main/java/org/reaktivity/nukleus/sse/internal/stream/SseServerFactory.java

```diff
@@ -318,35 +319,35 @@ public MessageConsumer newInitialSseStream(
 
         // TODO: need lightweight approach (end)
 
-        final MessagePredicate filter = (t, b, o, l) ->
-        {
-            final RouteFW route = routeRO.wrap(b, o, o + l);
-            final SseRouteExFW routeEx = route.extension().get(sseRouteExRO::tryWrap);
-            final String routePathInfo = routeEx != null ? routeEx.pathInfo().asString() : null;
-
-            // TODO: process pathInfo matching
-            //       && pathInfo.startsWith(routePathInfo);
-            return true;
-        };
-
         MessageConsumer newStream = null;
 
-        final RouteFW route = router.resolve(acceptRouteId, authorization, filter, wrapRoute);
-        if (route != null)
+        if (lastEventIdValid)
```

Comment:
We only need to validate the last event id we are planning to use, without impacting precedence rules of header over query parameter.

Then since we're only doing it once, the method can be called directly here.

Then the method call can be inlined to read
```
if (lastEventId == null || lastEventId.length() <= MAXIMUM_LAST_EVENT_ID_SIZE)
```

Reply:
The reason why i did it that way is that if the header is already wrong we should be going into 

```
 // extract lastEventId query parameter from pathInfo
        // use query parameter value as default for missing Last-Event-ID header
        if (pathInfo
```